### PR TITLE
fix: [CDS-39946]: infrastructure not updating on change of environment

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
### Summary

Infrastructure not updating on change of environment

#### Screenshots

https://user-images.githubusercontent.com/96711906/180153626-4c47a7ab-c398-44a7-8a27-2d5f7423abd1.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/DeployInfrastructures.tsx [hunk 1]
```diff
@@ -212,12 +217,13 @@ function DeployInfrastructures({
...
 
   useEffect(() => {
+    // istanbul ignore else
     if (!isNil(infrastructuresError)) {
       showError(getRBACErrorMessage(infrastructuresError))
     }
   }, [infrastructuresError])
 
-  const updateInfrastructuresList = (values: InfrastructureResponseDTO) => {
+  const updateInfrastructuresList = /* istanbul ignore next */ (values: InfrastructureResponseDTO) => {
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/DeployInfrastructures.tsx:226]: Why `istanbul ignore next` is added here ?

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/__tests__/DeployInfrastructures.test.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+import { TestWrapper } from '@common/utils/testUtils'
+
+import DeployInfrastructures from '../DeployInfrastructures'
+
+import mockInfrastructures from './__mocks__/mockInfrastructures.json'
+import mockInfrastructureInputs from './__mocks__/mockInfrastructureInputs.json'
+
+const showError = jest.fn()
+
+jest.mock('@harness/uicore', () => ({
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/__tests__/DeployInfrastructures.test.tsx:27]: We must not do this.

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/__tests__/DeployInfrastructures.test.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            allowableTypes={[MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME]}
+            initialValues={{
+              infrastructureRef: 'infra1'
+            }}
+          />
+        </Formik>
+      </TestWrapper>
+    )
+
+    expect(container).toMatchSnapshot()
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/__tests__/DeployInfrastructures.test.tsx:124]: No need to write snapshot testing here. Simple assertion would do just fine.
Snapshot testing is causing lot of issues when even simple style is changed in common component.

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/__tests__/DeployInfrastructures.test.tsx [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            allowableTypes={[MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME]}
+            initialValues={{
+              infrastructureRef: ''
+            }}
+          />
+        </Formik>
+      </TestWrapper>
+    )
+
+    expect(container).toMatchSnapshot()
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/__tests__/DeployInfrastructures.test.tsx:162]: Same as above.

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/__tests__/DeployInfrastructures.test.tsx [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+            allowableTypes={[MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME]}
+            initialValues={{
+              infrastructureRef: ''
+            }}
+          />
+        </Formik>
+      </TestWrapper>
+    )
+
+    await waitFor(() => expect(showError).toHaveBeenCalled())
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/__tests__/DeployInfrastructures.test.tsx:203]: Assert on actual error message here. And remove mocking of showError.

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/DeployInfrastructures.tsx (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructures/DeployInfrastructures.tsx:226]: I'll test this with cypress

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by vkbansal — APPROVED** [review]

**Pull request merged**